test/bin/build_images.sh

```diff
@@ -376,12 +377,47 @@ do_group() {
         done
     fi
 
-    if (( ${#buildid_list[@]} )); then
+    if ${BUILD_INSTALLER} && ! ${COMPOSER_DRY_RUN}; then
+        for download_file in "${groupdir}"/*.image-fetch; do
+            local download_url
+            download_url=$("${GOMPLATE}" --file "${download_file}")
+            blueprint="$(basename -s .image-fetch "${download_file}")"
+            local expected_iso_file="${VM_DISK_BASEDIR}/${blueprint}.iso"
+            if [ -f "${expected_iso_file}" ]; then
+                echo "${expected_iso_file} already exists"
+                if should_skip "${blueprint}"; then
+                    record_junit "${groupdir}" "${download_file}" "download" "SKIPPED"
+                    continue
+                fi
+            fi
+
+            echo "Starting image-fetch for ${download_file}"
+            wget -nv -O "${expected_iso_file}" "${download_url}" &
+            download_pids+=($!)
+        done
+        # Make sure all image-fetch processes are terminated on exit
+        if [ ${#download_pids[@]} -ne 0 ]; then
+            # Expand the values of download PIDs here not to reference
+            # the local variable on script exit
+            # shellcheck disable=SC2064
+            trap "kill ${download_pids[*]}" EXIT
```

Comment:
Does this override any other exit trap, like the one to collect logs?

Reply:
If I use parallel, this one should not be necessary, I think. 
I'll upload the new code asap.